functions/_fzf_search_completions.fish

```diff
@@ -22,15 +25,20 @@ function _fzf_search_completions --description "Shell completion using fzf"
 	# want to review first (e.g. man ima<TAB> might match `feh`, an image viewer)
 	if test (count $completions) -eq 1; and test (string sub -l (string length -- (commandline -ct)) -- $first_completion) = (commandline -ct)
 		# If there is only one option then we don't need fzf
-		# Everytime we need to cut, we need zero termination to
-		# avoid newlines from being interpreted later.
 		set results $first_completion
 	else
-		# We preprocess the whole list to prevent spawning a slow cut process for each completion.
-		# From here on, make sure to use -- and zero termination everywhere to prevent introducing
+		# Preprocess the whole list and extract the actual completion any the corresponding description (if any)
+		# From here on, make sure to use -- (argument list end) and zero termination everywhere to prevent introducing
 		# new edge-cases where characters are interpreted wrongly
-		set -l actual_completions (string join0 -- $completions | cut -f1 --zero-terminated | string split0)
-		set -l descriptions (string join0 -- $completions | cut -f2- --zero-terminated | string split0)
+		set -l actual_completions
+		set -l descriptions
+		for i in (seq (count $completions))
+			# split on \t from the right, at most one time and use the first (left) field.
+			# This is the actual completion
+			set actual_completions[$i] (string split --fields 1 --max 1 --right \t -- $completions[$i])
+			# The other field is the description, if it exists (otherwise set empty).
+			set descriptions[$i] (string split --fields 2 --max 1 --right \t -- $completions[$i] ; or echo "")
+		end
```

Comment:
you can actually do something like
```fish
for completion in $completions
    set --append actual (string split $completion)
    set --append description (string split $completion)
```
to make this part easier to read :)

Reply:
Done